python_modules/libraries/dagster-aws/dagster_aws/pipes/message_readers.py

```diff
@@ -216,11 +217,34 @@ def no_messages_debug_text(self) -> str:
         )
 
 
+# Number of retries to attempt getting cloudwatch logs when faced with a throttling exception.
+DEFAULT_CLOUDWATCH_LOGS_MAX_RETRIES = 10
+
+def get_log_events(client: "CloudWatchLogsClient", max_retries: Optional[int] = None, **log_params):
+    max_retries = max_retries or DEFAULT_CLOUDWATCH_LOGS_MAX_RETRIES
+
+    def get_log_events_delay_generator() -> Iterator[float]:
```

Comment:
let's place this `def` in the top-level module scope

Reply:
Done